server/helpers/pagination.js

```diff
@@ -1,11 +1,11 @@
 export const generatePagination = (page, limit, total) => {
   const currentPage = parseInt(page, 10) < 1 ? 1 : parseInt(page, 10);
   return {
-    currentPage,
+    currentPage: limit === 0 ? 1 : currentPage,
     limit: parseInt(limit, 10),
     offset: (currentPage - 1) * limit,
     total,
-    totalPage: Math.ceil(total / limit) || 0,
+    totalPage: limit === 0 ? 1 : Math.ceil(total / limit) || 0,
```

Comment:
Same here:

```suggestion
    totalPage: removeLimit ? 1 : Math.ceil(total / limit) || 0,
```

Also, we might want to consider the value of `totalPage` if limit is 0 and total is 0.